holoviews/plotting/mpl/util.py

```diff
@@ -185,3 +186,33 @@ def get_raster_array(image):
         else:
             data = np.flipud(data)
     return data
+
+def ring_coding(array):
```

Comment:
Is this called `ring` because it is closed (or at least looks closed)? If so the docstring should mention this, if not, I'm not sure what 'ring' is referring to. If I remember right, matplotlib has explicit codes for declaring genuinely closed paths that I assume you've considered using...

Reply:
It generates the path codes for each of the exterior and interior paths (which are always rings in a polygon). Using CLOSEPOLY path codes is not necessary since PathPatch will automatically close any patches.